TROUBLESHOOTING.md

```diff
@@ -2,13 +2,116 @@
 
 ## Table of contents
 
-- [How to enable logging](#how-to-enable-logging)
-- [Common errors](#common-errors)
-  - [Login does not work](#login-does-not-work)
-  - [Entity Not Found error](#entity-not-found-error)
-  - [vcfa_content_library_item creation never finishes (it's stuck)](#vcfa_content_library_item-creation-never-finishes-its-stuck)
+- [About Terraform](#about-terraform)
+  - [State management](#state-management)
+  - [Terraform native logging](#terraform-native-logging)
+  - [Provider version constraints](#provider-version-constraints)
+  - [Terraform architecture and responsibility boundaries](#terraform-architecture-and-responsibility-boundaries)
+    - [Terraform Core responsibilities](#terraform-core-responsibilities)
+    - [Terraform Provider plugin responsibilities](#terraform-provider-plugin-responsibilities)
+    - [Provider version constraints](#provider-version-constraints)
+- [Terraform provider VCFA](#about-terraform)
+  - [How to enable logging](#how-to-enable-logging)
+  - [Common errors](#common-errors)
+    - [Login does not work](#login-does-not-work)
+    - [Entity Not Found error](#entity-not-found-error)
+    - [vcfa_content_library_item creation never finishes (it's stuck)](#vcfa_content_library_item-creation-never-finishes-its-stuck)
+    - [External entity connection (vCenter server, NSX Manager) returns certificate error](#external-entity-connection-vcenter-server-nsx-manager-returns-certificate-error)
 
-## How to enable logging
+## About Terraform
+
+This section briefly touches on *Terraform* functionality that is provided within
+[`terraform`][terraform] binary by [Hashicorp][hashicorp].
+
+*Note:* For the VCFA provider specific logging see [How to enable logging](#how-to-enable-logging).
+
+### State management
+
+One of core features of Terraform is storing [infrastructure state][terraform-state]. The state can
+either be stored [locally, or remotely][state-storage]. The state contains mapping of existing
+infrastructure and it serves multiple purposes: drift detection, performance, additional metadata
+tracking.
+
+Local state is usually stored in `terraform.tfstate` file, while remote state storage specifics depend on
+[state backend][state-backend] being used. `terraform state` subcommand provides CLI capabilities
+for state management.
+
+*Note:* Terraform state may contain [sensitive data][state-sensitive-data].
+
+### Terraform native logging
+
+Terraform itself, independently of the provider plugin being used, has some debugging options that
+can enable additional logging. They are documented in [Terraform debugging
+page][terraform-debugging].
```

Comment:
I wouldn't mind those env variables just listed here for simplicity (how to enable the logging).

Reply:
Added